### ISSUE TITLE
FRI Session could not be established

### ISSUE THREAD

**Issue by adamleon** [lbr-stack/lbr_fri_ros2_stack#141]
Hello, I'm trying to set up our LBR Med 14 robot, but I cannot establish the FRI session.

![2023-12-05 12 02 22](https://github.com/lbr-stack/lbr_fri_ros2_stack/assets/1880042/5743bba2-54e5-4bd4-b7f9-5887142b90ef)

I can ping the robot on 172.31.1.147, so I guess there is a connection. But it immediatly shuts down when I try to run the server.

My setup is Humble on 22.04. And I'm running
```
ros2 launch lbr_bringup bringup.launch.py model:=med14 sim:=false rviz:=true moveit:=false
```

I saw the error is similar to [this issue](https://github.com/lbr-stack/lbr_fri_ros2_stack/issues/24), but I cannot resolve it in the same way.

Thanks in advance

**Comment by mhubii** [comment]
hi @adamleon , what version of the FRI are you on?

Is your computers IP configured to:

IP: `172.31.1.148`
Netmask: `255.255.0.0`

**Comment by adamleon** [comment]
172.31.1.148
255.255.255.0

Wow, that was a fast response

**Comment by mhubii** [comment]
no problem, was just working on something. Can you try to update your netmask?

**Comment by adamleon** [comment]
I updated it to 255.255.0.0 and it still failed if that was what you meant.

**Comment by mhubii** [comment]
okay 2 other possible causes:

- there is a chance that your FRI version is slightly different
  - could you please extract the FRI library through workbench as outlined here https://github.com/lbr-stack/fri#contributing
- other users had to open the required ports: https://github.com/lbr-stack/lbr_fri_ros2_stack/issues/67#issuecomment-1479794623

**Comment by adamleon** [comment]
I checked my FRI version and it's 1.15.1.SMedV121-15-0-35. If this is a version you want me to extract, then I'll do it

I tried to open the ports with the command spesified in the issue, but it didn't help

**Comment by adamleon** [comment]
Hmm... I tired to save the version as outlined, however I get the error:
![Skjermbilde 2023-12-05 131939](https://github.com/lbr-stack/lbr_fri_ros2_stack/assets/1880042/ea17f781-2785-4b78-a619-f0c41bd74bc1)

Could this be an issue?

**Comment by mhubii** [comment]
this indicates that you'll have to add your robot in the `StationSetup.cat` -> `Topology` tab before you do something else

maybe follow the robot setup first https://lbr-fri-ros2-stack-doc.readthedocs.io/en/latest/lbr_fri_ros2_stack/lbr_fri_ros2_stack/doc/robot_setup.html

**Comment by adamleon** [comment]
I fixed the issue of the Topology version. Seemed like it was set to Version 2. Setting it to Version 1, made me able to save the FRI version.

I have followed the robot setup. I ran into a few issues with Java versions, since I got errors in the LBRServer file when running on the default jdk. Changed the jdk to jdk1.7.0_80 and the compiler to 1.6, which somehow fixed the issue and I was able to upload the project

**Comment by mhubii** [comment]
so in that case, could it be that the LBRServer java application does not work with your version?

**Comment by adamleon** [comment]
Could be. The error I got was:
```
The type java.io.ObjectInputStream cannot be resolved. It is indirectly referenced from required .class files
```
after some Googling it seemed to be a compatibility issue with breaks at Java 1.8 and up. And I don't know what Java version my robot should be running on.

**Comment by mhubii** [comment]
it might make sense for you to try and run KUKA's FRI demos first, i.e. without any ROS 1/2 dependency.

You might want to run the LBRJointSineOverlay java application and the corresponding FRI client side application

- On smartPAD:
![applications_joint_sine_overlay1](https://github.com/lbr-stack/lbr_fri_ros2_stack/assets/26366414/501fb28a-e81e-49d7-8b8f-ffad8531e181)
- On your computer
https://github.com/lbr-stack/fri#run-the-apps

On another note, did you execute a brake test prior to establishing a connection?

**Comment by adamleon** [comment]
I did that now. Seems I get the same error. This is probably an issue with the FRI on the robot. I've contacted KUKA Service to help with this. I'll let you know when this is resolved

**Comment by adamleon** [comment]
Thanks BTW. You've helped me alot!

**Comment by mhubii** [comment]
happy to help, please let me know how it goes!

**Comment by adamleon** [comment]
Hey! I found the solution. The solution was that I had not installed any packages in the Sunrise Project. 
When I ran the SineOverlay test, it worked.
However, the adventure did not end there. The LBRServer did not compile because of a compile error "The type java.io.ObjectInputStream cannot be resolved", which was solved by setting the JRE System Library to jdk.1.7.0_80 and the Compiler to 1.6
It has something to do with enum not being supported in some versions of Java.
